security/android_vbmeta.ksy

```diff
@@ -0,0 +1,172 @@
+        type: position
+        doc: Position of signature data in "Authentication data" block.
+      - id: public_key
+        type: position
+        doc: Position of public key in "Auxiliary data" block.
+      - id: public_key_metadata
+        type: position
+        doc: Position of public key metadata in "Auxiliary data" block.
+      - id: descriptors
+        type: position
+        doc: Position of descriptor data in "Auxiliary data" block.
+      - id: rollback_index
+        type: u8
+        doc: |
+          The rollback index which can be used to prevent rollback to
+          older versions.
+      - id: flags
+        type: u4
+        doc: |
+          Flags from the AvbVBMetaImageFlags enumeration. This must be
+          set to zero if the vbmeta image is not a top-level image.
+      - id: reserved0
+        contents: [0, 0, 0, 0]
+        doc: |
+          Reserved to ensure |release_string| start on a 16-byte boundary.
+          Must be set to zeroes.
+      - id: release_string
+        size: 48
+        type: strz
+        encoding: ASCII
+        doc: |
+          The release string from avbtool, e.g. "avbtool 1.0.0" or
+          "avbtool 1.0.0 xyz_board Git-234abde89". Is guaranteed to be NUL
+          terminated. Applications must not make assumptions about how this
+          string is formatted.
+      - id: reserved
+        size: 80
+        doc: |
+          Padding to ensure struct is size AVB_VBMETA_IMAGE_HEADER_SIZE
+          bytes. This must be set to zeroes.
+    instances:
+      is_hashtree_disabled:
+        value: flags & 1 != 0 
+      is_verification_disabled:
+        value: flags & 2 != 0
+  vbmeta_public_key_header:
+    doc: The header for a serialized RSA public key
+    seq:
+      - id: key_num_bits
+        type: u4
+        doc: The size of the key in bits
+      - id: n0inv
+        type: u4
+        doc: Precomputed value for optimization of verification.
+  vbmeta_descriptor:
+    seq:
+      - id: tag
+        type: u8
+        enum: descriptor_types
+      - id: len_data
+        type: u8
+      - id: data
+        size: len_data
+  vbmeta_descriptors:
+    seq:
+      - id: descriptors
+        type: vbmeta_descriptor
+        repeat: eos
+  vbmeta_authentication_data:
+    instances:
+      hash_body:
+        pos: _root.header.hash.offset
+        size: _root.header.hash.size
+      signature_body:
+        pos: _root.header.signature.offset
+        size: _root.header.signature.size
+  vbmeta_auxiliary_data:
+    instances:
+      public_key_body:
+        pos: _root.header.public_key.offset
+        size: _root.header.public_key.size
+        type: vbmeta_public_key_header
+      public_key_metadata_body:
+        pos: _root.header.public_key_metadata.offset
+        size: _root.header.public_key_metadata.size
+      descriptors_body:
+        pos: _root.header.descriptors.offset
+        size: _root.header.descriptors.size
+        type: vbmeta_descriptors
```

Comment:
Why do you add the `_body` suffix everywhere? Why not just `public_key`, `public_key_metadata`, `descriptors`?

Maybe it's just about using different names than the ones already in `vbmeta_header`:

https://github.com/kaitai-io/kaitai_struct_formats/blob/e877931eec95cc0b38b36609025cf2d568c96423/security/android_vbmeta.ksy#L70-L78

However, I'd do it the other way around: dropp the `_body` suffix for actual structs filled with data and add a suffix `_pos` (for example) to the fields in `header` that only provide position info.